red_queen/games/applications/grovers.py

```diff
@@ -0,0 +1,228 @@
+# ------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the Apache 2.0 License.
+# See accompanying file /LICENSE for details.
+# ------------------------------------------------------------------------------
+
+# Copyright 2022 SRI-International organization
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# github repository:
+# https://github.com/Lementknight/QC-App-Oriented-Benchmarks/blob/master/grovers/qiskit/grovers_benchmark.py
```

Comment:
Before this gets merged, make sure you site the original repository and not your fork.
```suggestion
# https://github.com/SRI-International/QC-App-Oriented-Benchmarks/blob/master/grovers/qiskit/grovers_benchmark.py
```

Reply:
On this, I typically recommend using a permalink for source code browser links like this so the link is to exactly the same version you were looking at. That way if things move around in a future revision of the upstream repository the link still works and points to the correct thing

Reply:
Oh, I probably should explain what a permalink is, see: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files

Reply:
I just added the permalink.